modulus/models/meshgraphnet/bsms_mgn.py

```diff
@@ -0,0 +1,163 @@
+
+from dataclasses import dataclass
+from typing import Iterable, List, Optional
+
+from dgl import DGLGraph
+from torch import Tensor
+
+from modulus.models.gnn_layers.bsms import BistrideGraphMessagePassing
+from modulus.models.meshgraphnet import MeshGraphNet
+from modulus.models.meta import ModelMetaData
+
+
+@dataclass
+class MetaData(ModelMetaData):
+    name: str = "BiStrideMeshGraphNet"
+    # Optimization, no JIT as DGLGraph causes trouble
+    jit: bool = False
+    cuda_graphs: bool = False
+    amp_cpu: bool = False
+    amp_gpu: bool = True
+    torch_fx: bool = False
+    # Inference
+    onnx: bool = False
+    # Physics informed
+    func_torch: bool = True
+    auto_grad: bool = True
+
+
+class BiStrideMeshGraphNet(MeshGraphNet):
+    """Bi-stride MeshGraphNet network architecture
+
+    Parameters
+    ----------
+    input_dim_nodes : int
+        Number of node features
+    input_dim_edges : int
+        Number of edge features
+    output_dim : int
+        Number of outputs
+    processor_size : int, optional
+        Number of message passing blocks, by default 15
+    mlp_activation_fn : Union[str, List[str]], optional
+        Activation function to use, by default 'relu'
+    num_layers_node_processor : int, optional
+        Number of MLP layers for processing nodes in each message passing block, by default 2
+    num_layers_edge_processor : int, optional
+        Number of MLP layers for processing edge features in each message passing block, by default 2
+    hidden_dim_processor : int, optional
+        Hidden layer size for the message passing blocks, by default 128
+    hidden_dim_node_encoder : int, optional
+        Hidden layer size for the node feature encoder, by default 128
+    num_layers_node_encoder : Union[int, None], optional
+        Number of MLP layers for the node feature encoder, by default 2.
+        If None is provided, the MLP will collapse to a Identity function, i.e. no node encoder
+    hidden_dim_edge_encoder : int, optional
+        Hidden layer size for the edge feature encoder, by default 128
+    num_layers_edge_encoder : Union[int, None], optional
+        Number of MLP layers for the edge feature encoder, by default 2.
+        If None is provided, the MLP will collapse to a Identity function, i.e. no edge encoder
+    hidden_dim_node_decoder : int, optional
+        Hidden layer size for the node feature decoder, by default 128
+    num_layers_node_decoder : Union[int, None], optional
+        Number of MLP layers for the node feature decoder, by default 2.
+        If None is provided, the MLP will collapse to a Identity function, i.e. no decoder
+    aggregation: str, optional
+        Message aggregation type, by default "sum"
+    do_conat_trick: : bool, default=False
+        Whether to replace concat+MLP with MLP+idx+sum
+    num_processor_checkpoint_segments: int, optional
+        Number of processor segments for gradient checkpointing, by default 0 (checkpointing disabled)
```

Comment:
@mnabian @Alexey-Kamenev would be nice to add some recommendation here as to what values one can use. When to use this (memory saving) etc. For this architecture and vanilla MGN too. Especially since we use this flag quite often during scaling. Any thoughts?

Reply:
Yes, that's a good point. At least, outline the general approach, such as "if your model does not fit into GPU memory, then start with the smallest factor of number of processor layers. For example, if the number of layers is 15, then you can start with 3 and increase in case the model still does not fit."

Not sure what is the best place to add it though - is it docstring or maybe docs?

Reply:
Yeah, maybe for now you can add it to the Modulus model docs here: https://docs.nvidia.com/deeplearning/modulus/modulus-core/api/modulus.models.html

Or just have it in the docstrings. Eventually we need to add a Perf tuning section to this repo similar to what we have for Sym. But that can be done in a separate PR :)